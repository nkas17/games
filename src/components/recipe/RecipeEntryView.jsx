import React from 'react';
import PropTypes from 'prop-types';
import TextInput from '../common/TextInput';
import SelectInput from '../common/SelectInput';

const RecipeEntryView = ({ recipe, categories, onSave, onChange, onCancel, saving, errors }) => (
	<form>
		<TextInput
			name="title"
			label="Title"
			value={recipe.title}
			onChange={onChange}
			error={errors.title}
		/>

		<TextInput
			name="description"
			label="Description"
			value={recipe.description}
			onChange={onChange}
			error={errors.description}
		/>

		<SelectInput
			name="category"
			label="Category"
			value={recipe.category}
			defaultOption="select category"
			options={categories}
			onChange={onChange}
			error={errors.category}
		/>

		<button
			id="save"
			type="submit"
			disabled={saving}
			className="btn btn-primary"
			onClick={onSave}
		>
			{saving ? 'saving...' : 'save'}
		</button>

		<button
			id="cancel"
			type="button"
			disabled={saving}
			className="btn btn-secondary"
			onClick={onCancel}
		>
			{'cancel'}
		</button>
	</form>
	);

RecipeEntryView.defaultProps = {
	saving: false,
	errors: {},
};

RecipeEntryView.propTypes = {
	recipe: PropTypes.objectOf(PropTypes.any).isRequired,
	categories: PropTypes.arrayOf(PropTypes.objectOf).isRequired,
	onSave: PropTypes.func.isRequired,
	onChange: PropTypes.func.isRequired,
	onCancel: PropTypes.func.isRequired,
	saving: PropTypes.bool,
	errors: PropTypes.objectOf(PropTypes.any),
};

export default RecipeEntryView;
