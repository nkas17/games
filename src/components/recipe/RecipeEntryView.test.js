import React from 'react';
import renderer from 'react-test-renderer';
import { shallow } from 'enzyme';
import RecipeEntryView from './RecipeEntryView';

/* eslint-disable no-undef */

const setup = saving => (
	{
		recipe: {},
		saving,
		errors: {},
		categories: [],
		onSave: () => {},
		onChange: () => {},
		onCancel: () => {},
	}
);

const enzymeSetup = saving => shallow(<RecipeEntryView {...setup(saving)} />);

describe('RecipeEntryView', () => {
	it('renders correctly', () => {
		const tree = renderer.create(
			<RecipeEntryView
				{...setup(false)}
			/>).toJSON();
		expect(tree).toMatchSnapshot();
	});

	it('has correct caption on save button when not saving', () => {
		const wrapper = enzymeSetup(false);
		expect(wrapper.find('#save').props().children).toBe('save');
	});

	it('has correct caption on save button when saving', () => {
		const wrapper = enzymeSetup(true);
		expect(wrapper.find('#save').props().children).toBe('saving...');
	});
});
