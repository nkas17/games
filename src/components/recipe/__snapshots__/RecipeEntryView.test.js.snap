// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`RecipeEntryView renders correctly 1`] = `
<form>
  <div
    className="form-group"
  >
    <label
      htmlFor="title"
    >
      Title
    </label>
    <input
      className="form-control"
      name="title"
      onChange={[Function]}
      placeholder=""
      type="text"
      value=""
    />
    
  </div>
  <div
    className="form-group"
  >
    <label
      htmlFor="description"
    >
      Description
    </label>
    <input
      className="form-control"
      name="description"
      onChange={[Function]}
      placeholder=""
      type="text"
      value=""
    />
    
  </div>
  <div
    className="form-group"
  >
    <label
      htmlFor="category"
    >
      Category
    </label>
    <select
      className="form-control"
      name="category"
      onChange={[Function]}
      value=""
    >
      <option
        value=""
      >
        select category
      </option>
    </select>
    
  </div>
  <button
    className="btn btn-primary"
    disabled={false}
    id="save"
    onClick={[Function]}
    type="submit"
  >
    save
  </button>
  <button
    className="btn btn-secondary"
    disabled={false}
    id="cancel"
    onClick={[Function]}
    type="button"
  >
    cancel
  </button>
</form>
`;
